ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLvertex.cpp

```diff
@@ -106,6 +110,19 @@ namespace enigma {
 
 bool vbo_is_supported = false;
 
+void graphics_init_vbo_method() {
+  // we don't check for extensions until GLEW has been initialized
+  vbo_is_supported = GLEW_ARB_vertex_buffer_object;
+  // if the vbo extension is supported, but the GL version is old,
+  // we need to alias the buffer functions to the extension ones
+  if (GLEW_ARB_vertex_buffer_object == true && GLEW_VERSION_1_5 == false) {
+    glGenBuffers = glGenBuffersARB;
```

Comment:
Why are you assigning values in the GL namespace? Also, who guarantees that `GLEW_VERSION_1_5` will be defined? That's a macro; preprocessors are safe to check it, but not code, *because it might not exist*, in which case the preprocessor would treat it as zero. When that condition is true, are these values null? If so, lose the if statement and just set each of these if they're null. Otherwise, fix the if statement to use our own preprocessor.

Reply:
I am assigning them because the GLEW people say it's the recommended way.
https://sourceforge.net/p/glew/bugs/208/

You either have glGenBuffers (GL 1.5 or greater) or glGenBuffersARB (GL 1.4 with VBO extension) or you have nothing (must use vertex arrays). I have already written the code with the most former obviously (plain old glGenBuffers), so the obvious workaround is to load the ARB versions into the GL namespace when they are all that is available.

Reply:
I see, so `GLEW_VERSION_foo` means GL version foo is supported, not that GLEW is at that version. That's fine, then.

Reply:
Understood, and yeah, I thought the exact same thing myself the first time I read it. I'm comfortable for now with the comment I've placed since it mentions that "the GL version is old" when this condition is checked.